dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -128,13 +136,28 @@ public T createHosted(@ApiParam(hidden = true) @Auth User user,
         long l = getEntryDAO().create(entry);
         T byId = getEntryDAO().findById(l);
         elasticManager.handleIndexUpdate(byId, ElasticMode.UPDATE);
-        return byId;
+
+        URI requestURI = uriInfo.getRequestUri();
+        int heIndex = requestURI.toString().indexOf("/hostedEntry");
+        String uriStrPrefix = requestURI.toString().substring(0, heIndex);
+        String uriStr = uriStrPrefix + "/" + Long.toString(l);
+
+        URI uriEntity = null;
+        try {
+            uriEntity = new URI(uriStr);
+        } catch (URISyntaxException e) {
+            LOG.error("URI for hosted workflow or tool is malformed");
```

Comment:
What situations would cause this but would (should?) still create a valid hosted entry?